src/coreclr/jit/compiler.cpp

```diff
@@ -4953,9 +4951,29 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     compJitTelemetry.NotifyEndOfCompilation();
 #endif
 
-#if defined(DEBUG)
-    ++Compiler::jitTotalMethodCompiled;
-#endif // defined(DEBUG)
+    unsigned methodsCompiled = (unsigned)InterlockedIncrement((LONG*)&Compiler::jitTotalMethodCompiled);
+
+    if (JitConfig.JitDisasmSummary() && !compIsForInlining())
+    {
+        const int BUFSIZE            = 20;
+        char      osrBuffer[BUFSIZE] = {0};
+        if (opts.IsOSR())
+        {
+            // Tiering name already includes "OSR", we just want the IL offset
+            sprintf_s(osrBuffer, BUFSIZE, " @0x%x", info.compILEntry);
+        }
+
+#ifdef DEBUG
+        const char* fullName = info.compFullName;
+        unsigned    hash     = info.compMethodHash();
+#else
+        const char* fullName  = eeGetMethodFullName(info.compMethodHnd);
+        unsigned    hash      = 0;
+#endif
+
+        printf("%4d: JIT compiled %s [%s%s, IL size=%u, code size=%u, hash=0x%08x %s]\n", methodsCompiled, fullName,
+               compGetTieringName(), osrBuffer, info.compILCodeSize, *methodCodeSize, hash, compGetStressMessage());
```

Comment:
`compGetStressMessage()` includes a leading space, so don't need one here.

```suggestion
        printf("%4d: JIT compiled %s [%s%s, IL size=%u, code size=%u, hash=0x%08x%s]\n", methodsCompiled, fullName,
               compGetTieringName(), osrBuffer, info.compILCodeSize, *methodCodeSize, hash, compGetStressMessage());
```